### PR TITLE
[neon2sse] create a new port

### PR DESCRIPTION
**Pull request by luncliff**
### What does your PR fix?  

Add a library `neon2sse`.  https://github.com/intel/ARM_NEON_2_x86_SSE
This project provides 1 header file and supports `find_package` for CMake projects.

```cmake
find_package(NEON_2_SSE CONFIG REQUIRED) # NEON_2_SSE::NEON_2_SSE
```

### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  

This is a header only. So every triplet must be available.

### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  

I think so

### CODE REVIEW

**Review by PhoebeHui — APPROVED** [review]
LGTM, thanks for the PR!

File: ports/neon2sse/fix-cmakelists.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index e4b8980..231b75e 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -2,7 +2,7 @@
+ # Modification: Copyright (C) 2017 David Hirvonen
+ 
+ cmake_minimum_required(VERSION 3.0)
+-project(NEON_2_SSE VERSION 1.0.0)
++project(neon2sse VERSION 1.0.0)
```
Comment (BillyONeal) [inline ports/neon2sse/fix-cmakelists.patch:10]: Can you add a comment somewhere in here describing why we are changing the project title?

Comment (BillyONeal) [reply ports/neon2sse/fix-cmakelists.patch:10]: (If you remove the patch, the `vcpkg_cmake_config_fixup` call must be changed to `vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/NEON_2_SSE)`)

**Comment by luncliff** [comment]
Indeed that would be great.

The project uses its name for config file generation.
So config fix up of `NEON_2_SSE` to `neon2sse` will break find_package with
neon2sse.

I thought matching port name and project looks better.

**Comment by BillyONeal** [comment]
Yes, I think it looks better too, but it also forces people who get this from vcpkg to use a different `find_package` name than the normal one upstream uses; ideally we match upstream exactly unless we have some reason we can't.

**Comment by luncliff** [comment]
Hmm... I will use package name 'NEON_2_SSE' instead.
 No patch file, no complexity. I agree.

**Comment by BillyONeal** [comment]
Thanks for the new port!

**Pull request merged**